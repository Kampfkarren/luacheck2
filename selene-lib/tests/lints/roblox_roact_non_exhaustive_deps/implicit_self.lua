local React

local function Component()
    local a = {}
    React.useEffect(function()
        local _ = a.b.c()
    end, { a.b.c })

    React.useEffect(function()
        -- Should require `a.b` as it is implicity passed
        local _ = a.b:c()
    end, { a.b.c })
end
