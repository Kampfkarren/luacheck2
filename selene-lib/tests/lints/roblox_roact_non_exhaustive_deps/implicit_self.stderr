error[roblox_roact_non_exhaustive_deps]: react hook useEffect has missing dependency: 'a.b'
   ┌─ implicit_self.lua:12:10
   │
12 │     end, { a.b.c })
   │          ^^^^^^^^^
   │
   = help: either include it or remove the dependency array

